packages/liveblocks-core/src/comments/types/ThreadData.ts

```diff
@@ -4,14 +4,14 @@ import type { CommentData } from "./CommentData";
 /**
  * Represents a thread of comments.
  */
-export type ThreadData<ThreadMetadata extends BaseMetadata = never> = {
+export type ThreadData<TThreadMetadata extends BaseMetadata = never> = {
```

Comment:
Not really feedback for this PR directly, but general. Would it be possible to rename this type to:

```suggestion
export type Thread<TMeta extends BaseMetadata = never> = {
```

I think that would be just as clear, yet much shorter. Or is `Thread` already taken as a type elsewhere?

Reply:
We've looked at this before and we couldn't make TypeScript accept both `Thread`/`Comment` types and `Thread`/`Comment` components. (And I think even if we did you get into issues quickly with auto-imports/ESLint etc?)

And regarding `TThreadMetadata`, I'm not a fan of naming the same generics/values differently throughout different places, e.g. having `TMeta` here but it's not possible in `createRoomContext` so `TThreadMeta` there and we refer to it in plain text as "thread metadata" because the property is called `metadata`. I think it gets confusing quickly and prevents easy understanding at a glance.

Reply:
> We've looked at this before and we couldn't make TypeScript accept both `Thread`/`Comment` types and `Thread`/`Comment` components. (And I think even if we did you get into issues quickly with auto-imports/ESLint etc?)

I see! Which of these types do will you be working with mostly as a user? For example, if the typical type you import from our library is this (`ThreadData`) type, then maybe we can export the component types as `ThreadComponent` instead, and rename this type to just `Thread`? I don't expect users to often use the `ThreadComponent` type in normal code, right? In that case, there is no conflict, and the most commonly needed type is nice and short.

> And regarding `TThreadMetadata`, I'm not a fan of naming the same generics/values differently throughout different places, e.g. having `TMeta` here but it's not possible in `createRoomContext` so `TThreadMeta` there and we refer to it in plain text as "thread metadata" because the property is called `metadata`. I think it gets confusing quickly and prevents easy understanding at a glance.

Oh, I see. Yeah, that makes sense if the same type variable appears elsewhere: better to keep those consistent in that case 👍

Reply:
> I see! Which of these types do will you be working with mostly as a user? For example, if the typical type you import from our library is this (ThreadData) type, then maybe we can export the component types as ThreadComponent instead, and rename this type to just Thread? I don't expect users to often use the ThreadComponent type in normal code, right? In that case, there is no conflict, and the most commonly needed type is nice and short.

Oh sorry, I meant that `Thread`/`Comment` are the actual components so even more useful than the `ThreadData`/`CommentData` types, which is why we decided to suffix the types.